src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -584,10 +584,17 @@ private static bool CanBindToTheseConstructorParameters(ParameterInfo[] construc
                 return;
             }
 
-            Type genericType = typeof(Dictionary<,>).MakeGenericType(keyType, valueType);
-
             MethodInfo tryGetValue = dictionaryType.GetMethod("TryGetValue")!;
```

Comment:
Should we use the same constraint as the `Item` property? e.g. `BindingFlags.Public | BindingFlags.Instance`?